src/libPersistence/ContractStorage2.cpp

```diff
@@ -277,6 +277,8 @@ bool ContractStorage2::FetchStateValue(const dev::h160& addr, const bytes& src,
     }
   }
 
+  delete it;
```

Comment:
There's some return statements above which will skip iterator deletion.

May I suggest to use something similar to what we do in [P2PComm](https://github.com/Zilliqa/Zilliqa/blob/ead49ba5d26b9526ab231de7de5afd42198fcbfc/src/libNetwork/P2PComm.cpp#L196)

For example:
```
void DeleteLevelDBIterator(leveldb::Iterator* it)
{
  delete it;
}

leveldb::Iterator* it = m_diagnosticDBNodes->GetDB()->NewIterator(leveldb::ReadOptions());
unique_ptr<leveldb::Iterator*, void (*)(leveldb::Iterator*)> ldbiter_deleter(it, DeleteLevelDBIterator);
```

This way we don't need to manually call `delete it` before every return statement.

Reply:
Simply replaced with unique_ptr for garbage collection